qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1500.py

```diff
@@ -0,0 +1,195 @@
+import textwrap
+from typing import Optional, Union
+from collections import defaultdict
+
+from qcodes import VisaInstrument
+from qcodes.utils.helpers import create_on_off_val_mapping
+from .KeysightB1530A import B1530A
+from .KeysightB1520A import B1520A
+from .KeysightB1517A import B1517A
+from .KeysightB1500_module import B1500Module, parse_module_query_response, \
+    parse_spot_measurement_response
+from . import constants
+from .constants import ChannelList
+from .message_builder import MessageBuilder
+
+
+class KeysightB1500(VisaInstrument):
+    """Driver for Keysight B1500 Semiconductor Parameter Analyzer.
+
+    For the list of supported modules, refer to :meth:`from_model_name`.
+    """
+
+    def __init__(self, name, address, **kwargs):
+        super().__init__(name, address, terminator="\r\n", **kwargs)
+
+        self.by_slot = {}
+        self.by_channel = {}
+        self.by_kind = defaultdict(list)
+
+        self._find_modules()
+
+        self.add_parameter('autozero_enabled',
+                           unit='',
+                           label='Autozero enabled of the high-resolution ADC',
+                           set_cmd=self._set_autozero,
+                           get_cmd=None,
+                           val_mapping=create_on_off_val_mapping(
+                               on_val=True, off_val=False),
+                           docstring=textwrap.dedent("""
+            Enable or disable cancelling of the offset of the 
+            high-resolution A/D converter (ADC).
+    
+            Set the function to OFF in cases that the measurement speed is 
+            more important than the measurement accuracy. This roughly halves
+            the integration time."""))
+        # Instrument is initialized with this setting having value of
+        # `False`, hence let's set the parameter to this value since it is
+        # not possible to request this value from the instrument.
+        self.autozero_enabled.raw_value = False
+        self.autozero_enabled._save_val(False)
+
+    def add_module(self, name: str, module: B1500Module):
+        super().add_submodule(name, module)
+
+        self.by_kind[module.MODULE_KIND].append(module)
+        self.by_slot[module.slot_nr] = module
+        for ch in module.channels:
+            self.by_channel[ch] = module
+
+    def reset(self):
+        """Performs an instrument reset.
+
+        This does not reset error queue!
+        """
+        self.write("*RST")
+
+    def get_status(self) -> int:
+        return int(self.ask("*STB?"))
+
+    # TODO: Data Output parser: At least for Format FMT1,0 and maybe for a
+    # second (binary) format. 8 byte binary format would be nice because it
+    # comes with time stamp
+    # FMT1,0: ASCII (12 digits data with header) <CR/LF^EOI>
```

Comment:
Is it ok to merge with this todo? It's not clear to me as someone not familiar with the instrument what limitations this gives.

Reply:
yes, that is OK. the improvements on the data format will come in the future PRs. for the use cases in the labs, this is enough so far.

I also don't like to leave TODOs in the code, but it's ok for now, because there are expected more PRs in the coming month or two which might clean most of those up.

Reply:
I'll comment on this, since I wrote this `TODO`: 
The B1500 provides many (I think >16) output formats, both ASCII and binary. The output format only tells how information (values, status messages, etc) is encoded in each item in the response. Additionally, what items are included depends on the measurement mode, and with which instruments modules the mode was configured.

My original plan was to create a parser for at least two of the formats that brings the response into some structured form. But due to the variety of result formats this is not trivial, which is why I went for the quick solution of having a separate parser for the mode that we currently use (which at the moment I think is just the Highspeed Spot IV measurement and the Highspeed spot Capacitance measurement.

If we ever want to make use of the other measurement modes I think it is still a good idea to pick this TODO up.

Reply:
thanks, Stefan, for clarification!